metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/EnrichmentWorkerBuilder.java

```diff
@@ -0,0 +1,124 @@
+  private String enrichmentUrl = null;
+  private int connectTimeout = 0;
+  private int responseTimeout = 0;
+
+
+  /**
+   * Set the URL of the dereferencing service. The default is null. If set to a blank value, the
+   * enrichment worker will not be configured to perform dereferencing.
+   *
+   * @param dereferenceUrl The URL of the dereferencing service.
+   * @return This instance, for convenience.
+   */
+  public EnrichmentWorkerBuilder setDereferenceUrl(String dereferenceUrl) {
+    this.dereferenceUrl = dereferenceUrl;
+    return this;
+  }
+
+  /**
+   * Set the URL of the enrichment service. The default is null. If set to a blank value, the
+   * enrichment worker will not be configured to perform enrichment.
+   *
+   * @param enrichmentUrl The URL of the dereferencing service.
+   * @return This instance, for convenience.
+   */
+  public EnrichmentWorkerBuilder setEnrichmentUrl(String enrichmentUrl) {
+    this.enrichmentUrl = enrichmentUrl;
+    return this;
+  }
+
+  /**
+   * Set the maximum amount of time, in milliseconds, we wait for a connection before timing out.
+   * The default (when not calling this method) is {@value EnrichmentWorkerBuilder#DEFAULT_CONNECT_TIMEOUT}
+   * milliseconds.
+   *
+   * @param connectTimeout The maximum amount of time, in milliseconds, we wait for a connection
+   * before timing out. If not positive, this signifies that the connection does not time out.
+   * @return This instance, for convenience.
+   */
+  public EnrichmentWorkerBuilder setConnectTimeout(int connectTimeout) {
+    this.connectTimeout = connectTimeout;
+    return this;
+  }
+
+  /**
+   * Set the maximum amount of time, in milliseconds, we wait for the response. The default (when
+   * not calling this method is {@value EnrichmentWorkerBuilder#DEFAULT_RESPONSE_TIMEOUT}
+   * milliseconds.
+   *
+   * @param responseTimeout The maximum amount of time, in milliseconds, we wait for the response
+   * before timing out. If not positive, this signifies that the response does not time out.
+   * @return This instance, for convenience.
+   */
+  public EnrichmentWorkerBuilder setResponseTimeout(int responseTimeout) {
+    this.responseTimeout = responseTimeout;
+    return this;
+  }
+
+  /**
+   * Builds an {@link EnrichmentWorker} according to the parameters that are set.
+   *
+   * @return An instance.
+   * @throws IllegalStateException When both the enrichment and dereference URLs are blank.
+   */
+  public EnrichmentWorker build() {
+
+    // Make sure that the worker can do something.
+    if (StringUtils.isBlank(dereferenceUrl) && StringUtils.isBlank(enrichmentUrl)) {
+      throw new IllegalStateException(
+              "Either dereferencing or enrichment (or both) must be enabled.");
+    }
+
+    // Create the request factory
+    final HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+    requestFactory.setConnectTimeout(Math.max(connectTimeout, 0));
```

Comment:
You need to set the default values here if there is none specified.. Seems that the defaults are not currently used

Reply:
Oops.. yes, i forgot to assign those to the right variables.